overlord/snapstate/snapmgr.go

```diff
@@ -416,10 +416,11 @@ func (m *SnapManager) doDownloadSnap(t *state.Task, _ *tomb.Tomb) error {
 
 	targetFn := snapsup.MountFile()
 	if snapsup.DownloadInfo == nil {
+		var storeInfo *snap.Info
 		// COMPATIBILITY - this task was created from an older version
 		// of snapd that did not store the DownloadInfo in the state
 		// yet.
-		storeInfo, err := theStore.Snap(snapsup.Name(), snapsup.Channel, snapsup.DevModeAllowed(), snapsup.Revision(), user)
+		storeInfo, err = theStore.Snap(snapsup.Name(), snapsup.Channel, snapsup.DevModeAllowed(), snapsup.Revision(), user)
```

Comment:
It's the follow up code not visible in the diff which is the problem. The err scoping will mean the err assignment below being ignored and our of scope by the time the check is made out of the branch.